File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/DrawingImageOrientation.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.annotations.SerializedName;
import java.io.IOException;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

/**
 * Gets or Sets Drawing.ImageOrientation
 */
@JsonAdapter(DrawingImageOrientation.Adapter.class)
public enum DrawingImageOrientation {
  TOPLEFT("TopLeft"),
  TOPRIGHT("TopRight"),
  BOTTOMRIGHT("BottomRight"),
  BOTTOMLEFT("BottomLeft"),
  LEFTTOP("LeftTop"),
  RIGHTTOP("RightTop"),
  RIGHTBOTTOM("RightBottom"),
  LEFTBOTTOM("LeftBottom");

  private String value;

  DrawingImageOrientation(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static DrawingImageOrientation fromValue(String input) {
    for (DrawingImageOrientation b : DrawingImageOrientation.values()) {
      if (b.value.equals(input)) {
        return b;
      }
    }
    return null;
  }

  public static class Adapter extends TypeAdapter<DrawingImageOrientation> {
    @Override
    public void write(final JsonWriter jsonWriter, final DrawingImageOrientation enumeration) throws IOException {
      jsonWriter.value(String.valueOf(enumeration.getValue()));
    }

    @Override
    public DrawingImageOrientation read(final JsonReader jsonReader) throws IOException {
      Object value = jsonReader.nextString();
      return DrawingImageOrientation.fromValue((String)(value));
    }
  }
}

File: SampleCode/RestApi/Net.RestSharp/EmbyClient.Dotnet/Model/DrawingImageOrientation.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Defines Drawing.ImageOrientation
    /// </summary>
    [JsonConverter(typeof(StringEnumConverter))]
        public enum DrawingImageOrientation
    {
        /// <summary>
        /// Enum TopLeft for value: TopLeft
        /// </summary>
        [EnumMember(Value = "TopLeft")]
        TopLeft = 1,
        /// <summary>
        /// Enum TopRight for value: TopRight
        /// </summary>
        [EnumMember(Value = "TopRight")]
        TopRight = 2,
        /// <summary>
        /// Enum BottomRight for value: BottomRight
        /// </summary>
        [EnumMember(Value = "BottomRight")]
        BottomRight = 3,
        /// <summary>
        /// Enum BottomLeft for value: BottomLeft
        /// </summary>
        [EnumMember(Value = "BottomLeft")]
        BottomLeft = 4,
        /// <summary>
        /// Enum LeftTop for value: LeftTop
        /// </summary>
        [EnumMember(Value = "LeftTop")]
        LeftTop = 5,
        /// <summary>
        /// Enum RightTop for value: RightTop
        /// </summary>
        [EnumMember(Value = "RightTop")]
        RightTop = 6,
        /// <summary>
        /// Enum RightBottom for value: RightBottom
        /// </summary>
        [EnumMember(Value = "RightBottom")]
        RightBottom = 7,
        /// <summary>
        /// Enum LeftBottom for value: LeftBottom
        /// </summary>
        [EnumMember(Value = "LeftBottom")]
        LeftBottom = 8    }
}
